Docs/6b_RuneRelationships.md

```diff
@@ -1,5 +1,254 @@
 # Rune: Relationships
 
+Relationships are an important part of an SQL database. Rune provides first class support for defining, keeping track of, and loading realationships between records.
+
+Out of the box, Rune supports three kinds of relationships, represented by property wrappers `@BelongsTo`, `@HasMany`, and `@HasOne`.
+
+## BelongsTo
+
+A `BelongsTo` is the simplest kind of relationship. It represents the child of a 1-1 or 1-M relationship. The child of a 1-1 or 1-M typically has a column referencing the primary key of another table.
+
+With Alchemy, you can define a belongs to relationship with the `@BelongsTo` property wrapper.
+
+```swift
+struct Person: Model {
+    static let tableName = "people"
+
+    var id: Int?
+    let name: String
+}
+
+struct Pet: Model {
+    static let tableName = "pets"
+
+    var id: Int?
+    let name: String
+
+    @BelongsTo
+    var owner: Person
+}
+```
+
+This signifies that a column on `Pet` references the id column of `Person`. By default, Alchemy will store the `id` type of `Person` (`Int`) into a column named `owner_id` on the `pets` table.
+
+Creating a new `Pet` would look like...
+
+```swift
+let owner: Person = ...
+let pet = Pet(
+    name: "Fido", 
+    owner: .init(owner) // Note that this is passing in a `BelongsTo` initialized containing a `Person`.
+)
+pet.save() // Save the pet; `owner.id` will be stored on the `owner_id` column of the `pets` table.
+```
+
+### Custom Column Suffix
+
+By default, `BelongsTo` relationships have the parent `Model`'s id mapped to a table column called `\(propertyName)Id`. Above, this was converted to `owner_id` because default `Model`s convert their properties to snake case.
+
+If you'd like to have a custom `BelongsTo` column suffix, you may do so by overriding the `Model.belongsToColumnSuffix` static property.
+
+```swift
+struct Pet: Model {
+    static let belongsToColumnSuffix = "ParentId"
+    ...
+}
+```
+
+This would result in the `owner` property being mapped to column `owner_parent_id`.
+
+## Eager Loading Relationships
+
+In order to access a relationship property of a queried `Model`, you need to load that relationship first. You can "eager load" it using the `.with()` function on a `ModelQuery`. Eager loading refers to preemptively, or "eagerly", loading a relationship before it is used.
+
```

Comment:
Might be good to add a section here about how it solved the N+1 problem here. Instead of having to do a query to say get pets and then another query for every pet to get their children, it becomes two queries that are then mapped. Much more efficient at scale.